pixeltable/iterators/document.py

```diff
@@ -106,14 +124,14 @@ def input_schema(cls) -> Dict[str, ColumnType]:
     def output_schema(cls, *args: Any, **kwargs: Any) -> Tuple[Dict[str, ColumnType], List[str]]:
         schema = {'text': StringType()}
         if 'metadata' in kwargs and len(kwargs['metadata']) > 0:
-            md_fields = kwargs['metadata'].split(',')
+            md_fields = [field.strip() for field in kwargs['metadata'].split(',')]
```

Comment:
Yuck, it just seems nasty and error prone that we're parsing this out a second time. And that we're using **kwargs here. Can we take the opportunity to refactor and do this right? Factor the `metadata` and `separators` parameter parsing out to avoid duplicated logic? and use typesafe method signatures when feasible? (I understand that some of these are legacy issues)